speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -306,6 +332,30 @@ static void LoadConfigSettings(void* p_param)
     // Set the max time setting from the xml file
     m_maxTime = std::stoi(GfParmGetStr(p_param, PRM_MAX_TIME, GFMNU_ATTR_TEXT, NULL));
 
+    const char* filePath = GfParmGetStr(p_param, PRM_BLACKBOX, "name", NULL);
+    if (filePath != nullptr) 
+    {
+        strcpy_s(m_blackBoxFilePath, BLACKBOX_PATH_SIZE, filePath);
+        SMediator* mediator = SMediator::GetInstance();
+        mediator->SetBlackBoxFilePath(m_blackBoxFilePath);
+        m_blackBoxChosen = true;
```

Comment:
Only the member variables should be set in this function (LoadConfig) The settings should be saved to the mediator in `SaveSettings()`